lib/light-service/organizer/iterate.rb

```diff
@@ -1,22 +1,82 @@
+# require 'light-service/action.rb'
+
 module LightService
   module Organizer
-    class Iterate
-      extend ScopedReducable
+    class Iterator
+      using ::LightService::Refinements::Array
+
+      include ScopedReducable
+
+      def initialize(collection_key, actions)
+        @collection_key = collection_key
+        @actions = actions
+        @iterations = nil
+        @current_iteration = nil
+      end
+
+      def rollback(ctx)
+        reversable_iterations.reverse.each_with_index do |single_iteration, idx|
+          ctx[single_iteration.item_key] = single_iteration.item
+
+          unless single_iteration.actions.include?(ctx.current_action)
```

Comment:
`single_iteration.actions.include?(ctx.current_action)` alias "Was the whole process interrupted while inside the iterator?"